pytorch_lightning/plugins/ddp_plugin.py

```diff
@@ -91,23 +93,31 @@ def init_ddp_connection(
                 torch_backend, rank=global_rank, world_size=world_size
             )
 
-    def on_before_forward(self, model: LightningModule, *args):
+    @property
+    def is_running_single_process_per_device(self) -> bool:
+        # objects do not need to be scattered in single process per device, move objects upfront to device
+        # used in `self.on_before_forward`
+        return self.device_ids is not None and len(self.device_ids) == 1
+
+    def on_before_forward(self, model: LightningModule, *batch):
         """
         Override to handle custom input to device logic. For DDP, no logic is required as this is handled internally
         within the DDP wrapper.
 
         Example::
 
-            def on_before_forward(self, model, *args):
-                batch, batch_idx = args
+            def on_before_forward(self, model, *batch):
+                batch, batch_idx = batch
                 return batch.to(model.device)
```

Comment:
```suggestion
                batch, batch_idx = batch
                return batch.to(model.device), batch_idx
```
btw it doesn't always contain only two values, can have `opt_idx`, `dataloader_idx`, `hiddens` as well. maybe generalize this example accordingly? Also, why rename it from args to batch?

Reply:
Done